macro/src/lib.rs

```diff
@@ -180,18 +180,40 @@ fn gen_custom_type_method(strct: &ast::CustomType, m: &ast::Method) -> Item {
         (quote! {}, quote! {})
     };
 
+    let writeable_flushes = m
```

Comment:
issue: this behavior needs doc updates

So it seems like `flush()` is already idempotent. We should document it as such in writeable.rs, that it must be possible to be called multiple times. We should allow it to ignore data after the first call.

We should also document that it gets called in the Diplomat extern code, and change the docs on DiplomatWriteable to not require that anymore

Reply:
I don't think it is idempotent, it adds null bytes. It *sounds* idempotent, it should probably be called `finalize` or something.

Reply:
Maybe it is because it doesn't append but insert it in a fixed position.

Reply:
Yes, it adds null bytes, but in a way that is idempotent: calling it twice does nothing.